src/components/ArticleDetails.jsx

```diff
@@ -21,11 +25,26 @@ export default function ArticleDetails() {
       console.log("id", res);
       if (res.status === 200) {
         setArticle(res.data.article);
+        setVotes(res.data.article.votes);
         getDate(res.data.article.created_at);
       }
     });
   }, []);
 
+  const updateVote = (updateType) => {
+    let noOfVotes = 0;
+    if (updateType === "increment") {
+      noOfVotes = 1;
+    } else if (updateType === "decrement") {
+      noOfVotes = -1;
+    }
```

Comment:
This is not quite optimistic rendering as your state for `votes` is set within a then block which is dependent on the api call.

The setting of the rendered votes needs to be independent of the api patch requests' success.

In short, you need to include `noOfVotes` in the returned render.

Reply:
I have added the change for optimistic rendering.